lib/perl/Genome/Model/ReferenceAlignment/Command/GenotypeMicroarrayConcordance.pm

```diff
@@ -252,28 +275,99 @@ sub resolve_genotype_microarray_vcf_and_sample {
-    );
+sub create_temp_file_path {
+    my $self = shift;
+    my $basename = shift;
 
-    # Check if we have seen this sample, and return the sorted vcf file name
-    my $sorted_microarray_vcf = $self->_genotype_samples_and_sorted_vcfs->{$genotype_sample->id};
-    return $sorted_microarray_vcf if $sorted_microarray_vcf;
+    if ($self->output_dir) {
+        return File::Spec->join($self->output_dir,$basename);
+    } else {
+        return Genome::Sys->create_temp_file_path($basename);
+    }
+}
 
-    # Create and store the sorted VCF file name for this sample
-    $sorted_microarray_vcf = Genome::Sys->create_temp_file_path(
-        Genome::Utility::Text::sanitize_string_for_filesystem($genotype_sample->name).'_microarray_sorted.vcf'
+sub resolve_builds {
+    my $self = shift;
+
+    my @builds;
+    if ($self->lane_qc_builds) {
+        my @instrument_data = $self->build->instrument_data;
+        unless (@instrument_data) {
+            $self->error_message('Found no instrument data assigned to build: '. $self->build->__display_name__);
+            die($self->error_message);
+        }
+        for my $instrument_data (@instrument_data) {
+            my $qc_build = $instrument_data->lane_qc_build;
+            unless ($qc_build) {
+                $self->error_message('Failed to find lane qc build for instrument data : '. $instrument_data->__display_name__);
+                die($self->error_message);
+            }
+            push @builds, $qc_build;
+        }
+    } else {
+        push @builds, $self->build;
+    }
+    return @builds;
+}
+
+sub intersect_vcfs {
+    my $self = shift;
+    my ($a_vcf,$b_vcf,$basename) = @_;
+    
+    my $raw_vcf = $self->create_temp_file_path($basename .'.vcf');
+    my $intersect_cmd = Genome::Model::Tools::BedTools::Intersect->create(
+        input_file_a => $a_vcf,
+        input_file_a_format => 'vcf',
+        input_file_b => $b_vcf,
+        output_file => $raw_vcf,
+        header => 1,
+        use_version => $self->bedtools_version,
     );
-    $self->_genotype_samples_and_sorted_vcfs->{$genotype_sample->id} = $sorted_microarray_vcf;
+    unless ($intersect_cmd) {
+        $self->error_message('Failed to create bedtools intersect!');
+        die($self->error_message);
+    }
+    unless ($intersect_cmd->execute) {
+        $self->error_message('Failed to execute bedtools intersect!');
+        die($self->error_message);
+    }
+    
+    my $sorted_vcf = $self->create_temp_file_path($basename .'.vcf.gz');
+    $self->sort_vcf($raw_vcf,$sorted_vcf);
 
-    return $sorted_microarray_vcf;
+    return $sorted_vcf;
+}
+
+sub resolve_roi_intervals_from_build {
+    my $self = shift;
+    my $build = shift;
+    
+    my $roi_bed = $self->create_temp_file_path($build->id .'.bed');
+    $build->region_of_interest_set_bed_file($roi_bed);
+
+    my $roi_intervals = $self->create_temp_file_path($build->id .'.intervals');
```

Comment:
I wish we would create tools that, if they want a filename to write a file to, would default to creating you a tempfile and then all you'd have to do is look at the command's output property for the path.  I imagine it would be easy to achieve this either with a UR property type or as a mixin class.  This pattern (create temp file, create command, execute command, return temp file's path) is used a lot inside of Genome.